### PR TITLE
KOA-4715 Update backpack-android version

### PR DESCRIPTION
**Pull request by marianeum**
Updated the project to the latest `backpack-android`. This includes changes to theming and a different publishing location. 

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)
+ [x] Any changes have been tested on both Android and iOS.

### CODE REVIEW

File: android/app/src/main/java/net/skyscanner/backpack/MainActivity.kt [hunk 1]
```diff
@@ -4,12 +4,10 @@ import android.content.res.Configuration
 import android.os.Bundle
 import com.facebook.react.ReactActivity
 import com.facebook.react.modules.i18nmanager.I18nUtil
-import net.skyscanner.backpack.util.BpkTheme
```
Comment (marianeum) [inline android/app/src/main/java/net/skyscanner/backpack/MainActivity.kt:7]: this is no longer needed - the theme provides everything needed

File: android/build.gradle [hunk 2]
```diff
@@ -20,7 +19,7 @@ buildscript {
 
 ext {
   internalBuild       = project.hasProperty("internalBuild") && project.internalBuild
-  backpackVersion     = "28.0.0${ext.internalBuild || isUseRelative() ? '-internal' : ''}"
+  backpackVersion     = "32.1.0"
```
Comment (marianeum) [inline android/build.gradle:22]: we no longer have separate versions

File: android/build.gradle [hunk 3]
```diff
@@ -54,22 +53,8 @@ allprojects {
           if (rootProject.file("local.properties").exists()) {
               localProperties.load(rootProject.file("local.properties").newDataInputStream())
           }
-
-          maven {
```
Comment (marianeum) [inline android/build.gradle:58]: we no longer need the artifactory version or jitpack as everything's on maven central

File: lib/android/src/androidTest/java/net/skyscanner/backpack/reactnative/testing/ReactContextTestRule.kt [hunk 4]
```diff
@@ -32,7 +32,7 @@ class ReactContextTestRule : TestWatcher() {
 
   override fun starting(description: Description) {
     context = ReactApplicationContext(InstrumentationRegistry.getInstrumentation().targetContext)
-    BpkTheme.applyDefaultsToContext(context)
+    context.setTheme(R.style.BpkTheme)
```
Comment (marianeum) [inline lib/android/src/androidTest/java/net/skyscanner/backpack/reactnative/testing/ReactContextTestRule.kt:35]: as we've just got a generic activity here we need to set the theme

File: lib/android/src/main/java/net/skyscanner/backpack/reactnative/rating/RNBpkRating.kt [hunk 5]
```diff
@@ -38,19 +38,19 @@ class RNBpkRating(
   }
 
   fun render() {
-    val view = getUpdatedView(state.orientation, state.size)
+    val view = getUpdatedView(state.orientation, state.size, state.scale)
```
Comment (marianeum) [inline lib/android/src/main/java/net/skyscanner/backpack/reactnative/rating/RNBpkRating.kt:41]: this is a new property that was added

**Review by olliecurtis — APPROVED** [review]
LGTM!

**Pull request merged**